src/System.Net.Http/src/System/Net/Http/HttpClientHandler.Unix.cs

```diff
@@ -160,14 +161,15 @@ protected override void Dispose(bool disposing)
 
         protected internal override Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
         {
-            return _curlHandler.SendAsync(request, cancellationToken);
+            return _curlHandlerPipeline.SendAsync(request, cancellationToken);
```

Comment:
I suggest that DiagnosticsHandler should own the logic to define if it's enabled or not at cost of one non-virtual static function 
In the part 2, it will start to send other events, so having this logic spread across all handlers would be quite hard to maintain.